nearcore/src/runtime/mod.rs

```diff
@@ -530,30 +530,23 @@ impl NightshadeRuntime {
                 ErrorKind::Other("Integer overflow during burnt balance summation".to_string())
             })?;
 
-        // Sort the receipts into appropriate outgoing shards.
-        let mut receipt_result = HashMap::default();
-        // Outgoing receipts should be sorted by shards of the epoch of the next block
-        let next_shard_layout = {
```

Comment:
This part of logic was wrong. Here we are getting the shard layout of the next block of "prev_block_hash", which is the shard layout of the current block. However, we need to get the shard layout of the next block after the current block. Since the current block info has not been stored at this point, it is very hard to get that info here

Reply:
It is also probably better in terms of code modularity to move the part where outgoing receipts are sorted by destination shard outside of runtime code since runtime should not care about shard layout at all, which should be handled on the chain level. (I know following the same logic, the code for splitting states here should also be moved to runtime, I just haven't have time to do so)

Reply:
Do we have a test which we can add to the old codebase which would surface this bug? Is such test added in this PR?

Reply:
This bug will only surface when shard layout changes. I haven't finished the full implementation of splitting shards, so we don't have end to end integration test for sharding upgrade now, but I will add that next week after I'm done with implementation